### PR TITLE
Remove global peak radius setting.

### LINKED ISSUE #6391: Make the peak radius an attribute of the function.
This issue was originally [TRAC 5545](http://trac.mantidproject.org/mantid/ticket/5545)

The PeakFunction has a default and quite small 'peak radius' - a cut-off  distance for the peak wings. It is a good thing for algorithms such as FindPeaks but can be a problem for a user who is unfamiliar with it. 

There should not be a global radius. It should be an attribute with different defaults for each peak type.

### PR DESCRIPTION
**Pull request by mantid-roman**
Rather than an attribute the `PeakRadius` is now a property of `Fit` (and some other algorithms) which I think is a much better option.

**To test:**

Fit some peaks (long tailed such as Lorentzians are preferred) with different values of `PeakRadius`. Plot the results, see how the property affects the cut-off region.

Fixes #6391.

Release notes have been updated.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by martyngigg — COMMENTED** [review]
Should we remove `curvefitting.peakRadius` from https://github.com/mantidproject/mantid/blob/master/Framework/Properties/Mantid.properties.template ?

File: Framework/API/src/IPeakFunction.cpp [hunk 1]
```diff
@@ -78,21 +80,28 @@ class TempJacobian : public Jacobian {
...
-  if (Kernel::ConfigService::Instance().getValue("curvefitting.peakRadius",
-                                                 peakRadius)) {
-    if (peakRadius != s_peakRadius) {
-      setPeakRadius(peakRadius);
-    }
+IPeakFunction::IPeakFunction(int peakRadius) : m_peakRadius(peakRadius) {}
+
+void IPeakFunction::function(const FunctionDomain &domain,
+                             FunctionValues &values) const {
+  auto peakRadius =
```
Comment (martyngigg) [inline Framework/API/src/IPeakFunction.cpp]: Why do we construct the `IPeakFunction` with a radius and then override it with the radius from the domain?

Comment (mantid-roman) [inline Framework/API/src/IPeakFunction.cpp]: The radius is overridden only when it's smaller than the default.

Comment (martyngigg) [inline Framework/API/src/IPeakFunction.cpp]: Could this cause confusion though if a user is trying to set it from `Fit` and then in some circumstances their value is not used?

Comment (mantid-roman) [inline Framework/API/src/IPeakFunction.cpp]: I think you are right. I'll make some changes.

File: Framework/CurveFitting/src/Algorithms/Fit.cpp [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 #include "MantidAPI/CostFunctionFactory.h"
 #include "MantidAPI/FuncMinimizerFactory.h"
+#include "MantidAPI/FunctionDomain1D.h"
```
Comment (martyngigg) [inline Framework/CurveFitting/src/Algorithms/Fit.cpp]: Is this new include required?

File: Framework/CurveFitting/src/Algorithms/Fit.cpp [hunk 3]
```diff
@@ -134,32 +135,16 @@ void Fit::execConcrete() {
     m_function->addConstraints(contstraints);
   }
 
-  // prepare the function for a fit
-  m_function->setUpForFit();
```
Comment (mantid-roman) [inline Framework/CurveFitting/src/Algorithms/Fit.cpp]: It was moved to the base class `IFittingAlgorithm`.

File: Framework/CurveFitting/src/Algorithms/Fit.cpp (hunk 2 above)
Comment (mantid-roman) [inline Framework/CurveFitting/src/Algorithms/Fit.cpp]: No. I forgot to remove it after I moved the peak radius stuff to the base class.

Comment (martyngigg) [inline Framework/CurveFitting/src/Algorithms/Fit.cpp]: Okay. If we remove this then I'm happy this can go through.

**Comment by martyngigg** [comment]
I've tried some fitting with a Voigt on some Vesuvio lead data. The peak radius settings seems to respond as expected and the fitting is quicker if I give it a positive, non-zero peak radius.

I think this is probably something not worth doing in this issue but I notice that setting a negative peak radius is allowed. I think users would be surprised at this and we should restrict to `>= 0`.

**Comment by martyngigg** [comment]
I'm happy with this now.

:shipit:

**Comment by SimonHeybrock** [comment]
@mantid-roman This branch has developed a merge conflict.

**Pull request merged**